src/main/java/clara/benchmark/interop/AccumRepeatedlyNotSelected.java

```diff
@@ -0,0 +1,50 @@
+package clara.benchmark.interop;
+
+import clara.rules.RuleLoader;
+import clara.rules.WorkingMemory;
+import clara.rules.QueryResult;
+import clara.benchmark.beans.Customer;
+import clara.benchmark.IBenchmark;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Iterable;
+
+public class AccumRepeatedlyNotSelected implements IBenchmark<WorkingMemory> {
+
+    //@Override
+    public String getName() {
+        return "clara.accum-repeatedly-not-selected";
+    }
+
+    //@Override
+    public WorkingMemory getSession() throws Exception {
+        return RuleLoader.loadRules("clara.benchmark.accum-repeatedly-not-selected");
+    }
+
+    //@Override
+    public Iterable<?> getFacts() throws Exception {
+        List<Customer> customers = new ArrayList<Customer>();
+        for (int i = 1; i < 100; i++) {
+            customers.add(new Customer(i, "CUSTOMER_NAME", true));
+        }
+        return customers;
+    }
+
+    //@Override
+    public void run(WorkingMemory session, Iterable<Object> facts) throws Exception {
+        WorkingMemory lastSession = session;
+
+        for (Object fact : facts) {
+            List<Object> toInsert = new java.util.ArrayList<Object>();
```

Comment:
I can change the getFacts method here to return an Iterable of Iterables.  The argument to toInsert does need to be an Iterable of facts to insert rather than a single fact since that is the signature of WorkingMemory [1], but you're right that this will save some time (although less than a millisecond if my trials just now at a REPL are reflective of the time used here).

I do think inserting one fact, firing, querying, and then repeating the cycle for each fact is what we want though.  The idea is to eliminate batch behavior in Clara (and any batch behavior in Drools; I'm not familiar enough with Drools internals to say what it does there).  I set this up so that the first fact inserted should be the one chosen by the accumulator.  The subsequent external insertions should not cause Rete operations downstream from the accumulating rule; if they do performance will suffer.  It would be possible to set a larger ruleset up to replicate this behavior, but it would be more complicated.   Aside from convenience here, I actually think this is a fair test for a rules engine.  Batch insertion isn't possible if an inherent feature of the problem space is that data is streaming over time and the program needs the state of the session in between the arrival of data points.



1. https://github.com/rbrush/clara-rules/blob/0.11.0/src/main/java/clara/rules/WorkingMemory.java#L16